pkg/util/cache.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+)
+
+type FileCache interface {
+	GetFile(CacheItem) (*os.File, error)
+	SetFile(CacheItem, *[]byte) error
+}
+
+type CacheItem struct {
+	FilePath string
+	URL      string
+	ShaURL   string
+}
+
+type DiskCache struct{}
```

Comment:
I like this level of indirection, but it's a bit awkward for callers to have to make a var cache util.DiskCache before calling any methods.

What do you think about adding a static cache to the top level of this package, and adding a SetCache function to allow swapping it out?

Similar to this pattern from libmachine: https://github.com/docker/machine/blob/master/libmachine/ssh/client.go